scripts/release

```diff
@@ -176,18 +169,29 @@ main() {
     fi
   fi
 
+  # Verify the latest commit has the version tag
+  # shellcheck disable=SC2155
+  local tag="$(git describe --exact-match HEAD)"
+  if [ "${tag}" != "${RELEASE_VERSION}" ]; then
+    log_error "Error: Expected HEAD to be tagged with ${RELEASE_VERSION}, but 'git describe --exact-match HEAD' reported: ${tag}"
+    exit 1
+  fi
+
+  # Verify the clean working tree
+  # shellcheck disable=SC2155
+  local diff="$(git diff --stat)"
```

Comment:
This might miss 2 types of the changes: 
  1 changes already staged for commit (`i.e. git diff --cached`) 
  2. .gitigored  files.

Reply:
I think we should check for 1.   
I think we can accept 2. as it's a minor risk to be source of inconsistency.

Reply:
Used `git diff HEAD` to catch both stages and unstages changes.
For 2 it would be worrying that files listed in gitignore could impact built release .

Reply:
(no HEAD) yet here.

Reply:
:P added https://github.com/etcd-io/etcd/pull/14051 on top

Reply:
Nice catch, we might need to use command below. I can cover both unstaged and staged changes, 
```
git status --short
```